### ISSUE TITLE
checkout_agreement_store doesn't exist

### ISSUE THREAD

**Issue by michelecanal** [magento/magento2#18357]
### Preconditions

1. Starting from Magento CE 2.2.5
2. upgrade from 2.2.5 to 2.2.6

### Steps to reproduce

1. Starting from Magento CE 2.2.5
2. created a terms and conditions (Store-->Terms and conditions)
3. upgrade from 2.2.5 to 2.2.6
3. The Terms and conditions created are not visible in admin (Store-->Terms and conditions)

### Expected result

1. I can edit the old terms and conditions
2. I can insert a new terms and conditions

### Actual result

1. old terms and conditions are not visible
2. In production mode I can add a new terms and conditions but is not visible in admin
3. In developer mode there is an exception: 

2 exception(s):
Exception #0 (Zend_Db_Statement_Exception): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'XXXXXX.checkout_agreement_store' doesn't exist, query was: SELECT `agreement_store`.* FROM `checkout_agreement_store` AS `agreement_store` WHERE (agreement_store.agreement_id IN ('1'))
Exception #1 (PDOException): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'XXXXXX.checkout_agreement_store' doesn't exist

The table exist in DB; note that in sql there isn't the prefix to table.

**Comment by magento-engcom-team** [comment]
Hi @michelecanal. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@michelecanal do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by michelecanal** [comment]
@magento-engcom-team give me test instance

**Comment by michelecanal** [comment]
HI,
I find the issue and the solution
In /vendor/magento/module-checkout-agreements/Model/ResourceModel/Agreement/Grid/Collection.php
There is an issue:

```
private function getStoresForAgreements()
    {
        $agreementId = $this->getColumnValues('agreement_id');

        if (!empty($agreementId)) {
            $select = $this->getConnection()->select()->from(
                ['agreement_store' => 'checkout_agreement_store']
            )->where(
                'agreement_store.agreement_id IN (?)',
                $agreementId
            );

            return $this->getConnection()->fetchAll($select);
        }

        return [];
    }
```
Change with
```
private function getStoresForAgreements()
    {
        $agreementId = $this->getColumnValues('agreement_id');

        if (!empty($agreementId)) {
            $select = $this->getConnection()->select()->from(
                [$this->getTable('agreement_store') => $this->getTable('checkout_agreement_store')]
            )->where(
                $this->getTable('agreement_store').'.agreement_id IN (?)',
                $agreementId
            );

            return $this->getConnection()->fetchAll($select);
        }

        return [];
    }
```

**Comment by sidolov** [comment]
Hi @michelecanal. Thank you for your report. 
The issue has been fixed in magento/magento2#18771 by @TomashKhamlai in 2.3-develop branch
Related commit(s):
 - [3cba00c951e0df4c843085ab761b9b97c2e3471b](https://github.com/magento/magento2/commit/3cba00c951e0df4c843085ab761b9b97c2e3471b)
 - [86e035f3ba615cb5e1bb3ff0e4821b82a57fe7fe](https://github.com/magento/magento2/commit/86e035f3ba615cb5e1bb3ff0e4821b82a57fe7fe)

The fix will be available with the upcoming 2.3.1 release.

**Comment by sidolov** [comment]
Hi @michelecanal. Thank you for your report. 
The issue has been fixed in magento/magento2#18412 by @pocallaghan in 2.3-develop branch
Related commit(s):
 - [2bee3660f183af1439b111557b3b958b3c812b25](https://github.com/magento/magento2/commit/2bee3660f183af1439b111557b3b958b3c812b25)

The fix will be available with the upcoming 2.3.1 release.

**Comment by magento-engcom-team** [comment]
Hi @michelecanal. Thank you for your report. 
The issue has been fixed in magento/magento2#18866 by @gelanivishal in 2.2-develop branch
Related commit(s):
 - [2a6f8340de5fd6fd96c4d0848ba84e9e64cc1cf0](https://github.com/magento/magento2/commit/2a6f8340de5fd6fd96c4d0848ba84e9e64cc1cf0)

The fix will be available with the upcoming 2.2.9 release.
